tempesta_fw/http.c

```diff
@@ -5603,7 +5603,12 @@ tfw_http_resp_cache(TfwHttpMsg *hmresp)
 	data.skb = NULL;
 	data.req = (TfwMsg *)req;
 	data.resp = (TfwMsg *)hmresp;
-	tfw_gfsm_move(&hmresp->conn->state, TFW_HTTP_FSM_RESP_MSG_FWD, &data);
+	if(tfw_gfsm_move(&hmresp->conn->state, TFW_HTTP_FSM_RESP_MSG_FWD, &data)) {
+		tfw_http_conn_msg_free(hmresp);
+		tfw_http_req_block(req, 403, "response blocked: filtered out");
```

Comment:
I see that there are other places with `tfw_http_conn_msg_free()` followed by `tfw_http_req_block()`, but I believe we don't need to call `tfw_http_conn_msg_free()` if request is paired with the response. The other similar places are: `tfw_http_resp_gfsm()`, `tfw_http_resp_terminate()`, and seems `tfw_http_resp_process()`.

Nothing really bad happens since `tfw_http_msg_free()` calls `tfw_http_msg_unpair()`, so there is no double freeing. But, `tfw_http_cli_error_resp_and_log()` frees the response as `req->pair` and there is just no reason to call the extra function and conditional statements inside it.

Please check my assumption and, it it's correct (**make an instrumentation patch insuring that the responses are really freed and there is no memory leak**), replace the `tfw_http_conn_msg_free()` calls preceding  `tfw_http_req_block()` with a comment like `/* The response is freed by tfw_http_req_block(). */`